### DISCUSSION TITLE
Struggling with Wasabi S3

### DISCUSSION THREAD

**Discussion by masterdanzh** [mastodon/mastodon#20813] [Q&A] [answered]
I installed Mastodon as Digitalocean Droplet and so far is looking good, beside an image issue...
Whenever I upload an image (avatar, server-thumbnail, ...) I see it saved in the bucket on Wasabi but the image is not shown on the Mastodon Website.

I've set policies both for the bucket and for the user but nothing seems to help.
And yes, object locking is enabled.

Any idea?

**Comment by supernovae** [comment, marked as answer]
Did you make your bucket public?

**Reply by masterdanzh** [reply]
Yep, I was missing some changes in the .env.production:

`S3_ALIAS_HOST=s3.wasabisys.com/[BUCKETNAME]`

and the Nginx conf:

```
#  location ~ ^/(emoji|packs|system/accounts/avatars|system/media_attachments/files) {
#    add_header Cache-Control "public, max-age=31536000, immutable";
#    add_header Strict-Transport-Security "max-age=31536000" always;
#    try_files $uri @proxy;
#  }

  location ~ ^/(emoji|packs) {
    add_header Cache-Control "public, max-age=31536000, immutable";
    try_files $uri @proxy;
  }

  location /system {
    rewrite ^/system(.*) https://s3.wasabisys.com/[BUCKETNAME]$1 permanent;
  }

```

**Comment by GreatScott** [comment]
@masterdanzh would you be willing to post the full S3 part of your Mastodon config? (with keys/bucket-name redacted, obviously)

I'm still having trouble here. Oddly, the site breaks for me with an error page whenever I save changes to an upload (rather than just displaying a broken image). I've verified everything works when not using S3 and hosting locally

I've made sure my Wasabi bucket is public and I've also attached policies to the bucket and Wasabi user according to: https://stanislas.blog/2018/05/moving-mastodon-media-files-to-wasabi-object-storage/

**Reply by GreatScott** [reply]
UPDATE. I finally got it working. Issue for me was that the region name needed to be included with the paths according to: https://wasabi-support.zendesk.com/hc/en-us/articles/360015106031-What-are-the-service-URLs-for-Wasabi-s-different-storage-regions-

```
S3_HOSTNAME=s3.us-central-1.wasabisys.com
S3_ENDPOINT=https://s3.us-central-1.wasabisys.com/
S3_BUCKET=[BUCKET_NAME]
S3_ALIAS_HOST=s3.us-central-1.wasabisys.com/[BUCKET_NAME]
```

**Reply by masterdanzh** [reply]
Sorry just saw your post now but happy that you nevertheless figured it out.
